### PR TITLE
plugins: allow staging to set all filter options

### PR DESCRIPTION
**Pull request by timmywil**

### CODE REVIEW

File: plugins/jquery-filters.php [hunk 1]
```diff
@@ -14,16 +14,20 @@
...
+	if ( !JQUERY_STAGING ) {
+		if ( $option === 'stylesheet' || $option === 'template' ) {
+			// Don't mess with themes for now.
+			continue;
+		}
+		if ( $option === 'active_plugins' ) {
+			// Deprecated
+			// In production, Puppet manages activation of per-site plugins.
+			// Locally, global plugins are enabled via mu-plugins/_loader.php,
+			// and per-site plugins are activated by the imported database.
```
Comment (Krinkle) [inline plugins/jquery-filters.php:28]: Should we remove the "Deprecated" and "Locally" phrasing here?

File: plugins/jquery-filters.php [hunk 2]
```diff
@@ -14,16 +14,20 @@
...
-		continue;
-	}
-	if ( $option === 'active_plugins' ) {
-		// Deprecated
-		// In production, Puppet manages activation of per-site plugins.
-		// Locally, global plugins are enabled via mu-plugins/_loader.php,
-		// and per-site plugins are activated by the imported database.
-		continue;
+	// Skip these in production, where they are managed by puppet.
+	// Staging should be allowed to set them for testing.
```
Comment (Krinkle) [inline plugins/jquery-filters.php:18]: I'd add that it is for testing "with an empty database" specifically, since it works fine without it in the way we've been testing so far.

Comment (timmywil) [reply plugins/jquery-filters.php:18]: I figure it should be allowed either way in case we ever wanted to test theme changes.

File: plugins/jquery-filters.php (hunk 1 above)
Comment (timmywil) [reply plugins/jquery-filters.php:28]: I agree with removing "Deprecated". "Locally" is still accurate tho, right?

File: plugins/jquery-filters.php (hunk 2 above)
Comment (Krinkle) [reply plugins/jquery-filters.php:18]: Right, I didn't think about the "allowed" phrasing. My point is that I foresee future "improvements" removing this because it seems to work. I want to leave a trace that says this enables primarily testing with a clean database, as it literally doens't work without it in that scenario. It can then also be used for other use cases indeed.

File: plugins/jquery-filters.php (hunk 1 above)
Comment (Krinkle) [reply plugins/jquery-filters.php:29]: No, because this code would no longer run locally. We're specifically making it so that per-site config is now determined by sites.php instead of by what is in the database.

Comment (timmywil) [reply plugins/jquery-filters.php:29]: I'm not sure I follow. "Locally" here describes the alternate code path, which is sites.php.

Comment (timmywil) [reply plugins/jquery-filters.php:29]: I see

**Review by Krinkle — APPROVED** [review]
